### DISCUSSION TITLE
Tuning Security Onion with THIS ONE WEIRD TRICK (BPF)!

### DISCUSSION THREAD

**Discussion by dougburks** [Security-Onion-Solutions/securityonion#6410] [Announcements]
https://www.youtube.com/watch?v=uamNOjtUi4Y

**Comment by danychouinard** [comment]
I love it.  This is a great idea.  But it doesn't work for me and I have no idea how to troubleshoot it.  The salt added so much abstraction that I'm completly lost.  

I've manage to understand the the spacing is critical.

but my salt-call is complaining :

  ID: pcap_state_not_allowed
  Function: test.fail_without_changes
  Result: False
  Comment: Failure!
  Started: 15:30:38.669840
  Duration: 2.314 ms
  Changes:

I have 5 machine in my grid.  I'm running it on the manager node.

I have two forward node, but only one is doing packet capture.  Thats the only thing I can think that is not vanilla setup.  

But the video was very clear and usefull, that's the kind of things that I need to keep going.  Simple and clear instruction that help me to grasp how things are done.  It will help me in the long run.  Congrat.

**Reply by InfosecGoon** [reply]
Glad you liked the video!

A couple of questions about your BPF configuration:

1. What file are you adding the BPF rules to? Is it your global.sls, or a minion file? If it's a minion file, is it for one of your forward nodes?
2. What are you trying to do with the BPF?
3. Can you post the configuration that you've added to the file so I can check the syntax?

Thanks!

--Matt

**Reply by danychouinard** [reply]
Hi Matt
I'm trying to replacate exaclty the demonstration.  I'm using the global.sls.
The hards drives of my server is having difficulties to keep up with the incoming packet.  Removing the 443 trafic sounded to me like quite a good idea.  It is at least giving my a chance to understand how things work.
In my file I don't think that I'm overdoing it much.  
    
```
    view /opt/so/saltstack/local/pillar/global.sls
   [...]
    115 docker:
    116   range: '172.17.0.0/24'
    117   bip: '172.17.0.1/24'
    118 redis_settings:
    119   redis_maxmemory: 812
    120 playbook:
    121   api_key: e2deb0203a87eb3137cde58d6e1514e6486d51d4
    122 steno:
    123   bpf:
    124     - not port 443


salt-call state.apply pcap
[ERROR   ] Failure!
local:
----------
          ID: pcap_state_not_allowed
    Function: test.fail_without_changes
      Result: False
     Comment: Failure!
     Started: 20:24:13.304124
    Duration: 2.49 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   2.490 msv
```

I wish I knew what it means

Dany

**Reply by InfosecGoon** [reply]
Dany -

I think there may be two issues here.

1. In the video, the configuration was being changed on a standalone installation. In your case, since you're using a distributed architecture, you want to put those steno configuration lines into the minion file for your forward node doing the packet capture. That's the only place that steno is running. The minion file should be at /opt/so/saltstack/local/pillar/minions/HOSTNAME_sensor.sls.

2. Similarly, since steno is running on the forward node, you're getting that pcap_state_not_allowed error because you're trying to restart it on the manager node. Instead, either SSH into the forward node and run the "salt-call state.apply pcap" command, or run it from the manager with "salt '*sensor' state.apply pcap" to restart it on your forward nodes.

Hope that helps!

--Matt
